lib/printer.rb

```diff
@@ -1,17 +1,32 @@
 class Printer
-  attr_reader :stdout, :grid
+  attr_reader :stdout
 
-  def initialize(stdout, grid)
+  def initialize(stdout)
     @stdout = stdout
-    @grid = grid
   end
 
   def print_welcome_message
-    stdout.puts(grid.template)
     stdout.puts("Enter your move >")
   end
 
+  def print_updated_grid(coordinates)
+    stdout.puts(print_current_grid(coordinates))
+  end
+
   def print_coordinates_error
     stdout.puts("Invalid input. Please try again.")
   end
+
+  def print_current_grid(coordinates)
```

Comment:
This method knows an in appropriate amount of stuff about the coordinates. It knows it is a hash, and we should really just be dealing with objects. Passing around data is an encapsulation smell that is better managed by a class.